### PR TITLE
MGMT-21204: add categories to feedback API

### PR DESCRIPTION
**Pull request by zszabo-rh**
## Description

This PR introduces structured, multi-select “quick response” feedback categories for negative (thumbs-down) user feedback and removes obsolete / positive categories.

Key points  
1. **New `FeedbackCategory` enum (11 values)** – negative-only categories (`incorrect`, `hallucinated_content`, …, `biased`, `unsafe`) with example comments for each.  
2. **`FeedbackRequest` model updated**  
   * Adds `categories: list[FeedbackCategory] | None`  
   * Validation ensures at least one of `sentiment`, `user_feedback`, or `categories` is provided.  
3. **Endpoint & docs**  
   * `/v1/feedback` OpenAPI schema + `docs/openapi.(json|md)` updated with new field, examples and category table.  
4. **Tests**  
   * All unit tests updated / added to cover the new enum, validation and endpoint behaviour (31 feedback-related tests now pass).  
5. **Docs sync** – `openapi.md` fully aligned with `openapi.json` for every endpoint description.  

## Type of change

- [ ] Refactor  
- [x] New feature  
- [ ] Bug fix  
- [ ] CVE fix  
- [ ] Optimization  
- [x] Documentation Update  
- [ ] Configuration Update  
- [ ] Bump-up service version  
- [ ] Bump-up dependent library  
- [ ] Bump-up library or tool used for development (does not change the final image)  
- [ ] CI configuration change  
- [ ] Konflux configuration change  
- [x] Unit tests improvement  
- [ ] Integration tests improvement  
- [ ] End to end tests improvement  

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

### 1. Unit tests
```bash
uv run pytest tests/unit/ -k "feedback" -v
```
All feedback-related tests pass.

### 2. Manual API verification
```bash
curl -X POST http://<host>:<port>/v1/feedback \
     -H "Content-Type: application/json" \
     -d '{
           "conversation_id":"<uuid>",
           "user_question":"What is the capital of France?",
           "llm_response":"Berlin",
           "sentiment":-1,
           "categories":["incorrect"]
         }'
```
Expected response: `{"response":"feedback received"}` (HTTP 200).  
Verify stored JSON file contains the new `categories` field.

### 3. Docs
Open `/docs` (Swagger UI) and confirm:  
* `FeedbackCategory` enum lists 11 entries with descriptions.  
* `FeedbackRequest` shows `categories` plus updated examples.  
* All endpoint descriptions match OpenAPI JSON.

### CODE REVIEW

File: src/models/requests.py [hunk 1]
```diff
@@ -213,9 +253,31 @@ def check_sentiment(cls, value: Optional[int]) -> Optional[int]:
...
+            return value
+        
+        if not isinstance(value, list):
+            raise ValueError("Categories must be a list")
+        
+        if len(value) == 0:
+            return None  # Convert empty list to None for consistency
+        
+        unique_categories = list(set(value))
+        return unique_categories
```
Comment (umago) [inline src/models/requests.py:270]: Should this validation take into account the "sentiment" field ? Because from L224 says: 

"List of feedback categories that describe issues with the LLM response (for negative feedback)."

So I assume, it's only required when feedback is negative.

File: tests/unit/models/test_requests.py [hunk 2]
```diff
@@ -232,3 +233,106 @@ def test_feedback_too_long(self) -> None:
...
+
+    def test_categories_invalid_type(self) -> None:
+        """Test validation error for invalid categories type."""
+        with pytest.raises(ValidationError):
+            FeedbackRequest(
+                conversation_id="123e4567-e89b-12d3-a456-426614174000",
+                user_question="Test question",
+                llm_response="Test response",
+                categories="invalid_type",  # Should be list, not string
+            )
```
Comment (umago) [inline tests/unit/models/test_requests.py:338]: We need a test with a positive review (sentiment) and an empty list of categories. Because from what I understood with this change, categories is only required when the feedback is negative.

**Review by umago — CHANGES_REQUESTED** [review]
Thanks, comments inline

File: tests/unit/models/test_requests.py (hunk 2 above)
Comment (umago) [reply tests/unit/models/test_requests.py:338]: Gotcha, thanks for the explanation

File: src/models/requests.py (hunk 1 above)
Comment (umago) [reply src/models/requests.py:270]: Gotcha, thanks for the explanation

**Review by umago — APPROVED** [review]
It LGTM, other teams should also look into this PR to see if it fits the expectations for their use of the feedback API.
